### PR TITLE
126 implement add experience

### LINKED ISSUE #126: Implement add experience

### PR DESCRIPTION
**Pull request by liadyacobi**
depends on #125 . Closes #126

### CODE REVIEW

File: android/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -12,6 +12,13 @@
     <string name="show_recent_experiences">Show recent experiences</string>
     <string name="no_experiences_for_movie">There are no experiences for \"%1$s\"</string>
 
+    <string name="post_experience">Post experience</string>
+    <string name="no_selection_placeholder">start by selecting a movie</string>
+    <string name="describe_experience_hint">describe your experience…</string>
+    <string name="title_experience_hint">give a title to your experience</string>
```
Comment (chenyair) [inline android/app/src/main/res/values/strings.xml:18]:
```suggestion
    <string name="post_experience">Post experience</string>
    <string name="no_selection_placeholder">Start by selecting a movie</string>
    <string name="describe_experience_hint">Describe your experience…</string>
    <string name="title_experience_hint">Give a title to your experience</string>
```

File: android/app/src/main/res/layout/fragment_add_experience.xml [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:padding="10dp">
+
+        <com.google.android.material.floatingactionbutton.FloatingActionButton
```
Comment (chenyair) [inline android/app/src/main/res/layout/fragment_add_experience.xml:23]: Make this icon like in MovieDetailsScreen and not fab

File: android/app/src/main/res/layout/fragment_add_experience.xml [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            android:layout_height="wrap_content"
+            android:text="@string/post_experience"
+            android:textAppearance="@style/TextAppearance.Material3.TitleMedium"
+            android:textColor="@color/md_theme_onSurfaceVariant"
+            app:layout_constraintBottom_toBottomOf="@+id/fabBack"
+            app:layout_constraintEnd_toStartOf="@+id/fabPost"
+            app:layout_constraintStart_toEndOf="@+id/fabBack"
+            app:layout_constraintTop_toTopOf="@+id/fabBack" />
+
+        <com.google.android.material.floatingactionbutton.FloatingActionButton
```
Comment (chenyair) [inline android/app/src/main/res/layout/fragment_add_experience.xml:49]: Make this icon also

File: android/app/src/main/res/layout/fragment_add_experience.xml [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+            android:backgroundTint="@color/md_theme_primary"
+            android:enabled="false"
+            android:src="@drawable/ic_done"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="@+id/fabBack"
+            app:shapeAppearanceOverlay="@style/RoundedCard"
+            app:tint="@color/md_theme_onPrimary" />
+
+        <com.google.android.material.card.MaterialCardView
+            style="?attr/materialCardViewElevatedStyle"
```
Comment (chenyair) [inline android/app/src/main/res/layout/fragment_add_experience.xml:65]:
```suggestion
            style="?attr/materialCardViewFilledStyle"
            app:cardBackgroundColor="@color/md_theme_surfaceContainer"
```

File: android/app/build.gradle.kts [hunk 5]
```diff
@@ -70,6 +70,7 @@ dependencies {
     implementation(libs.firebase.auth)
     implementation(libs.play.services.auth)
     implementation(libs.firebase.firestore)
+    implementation(libs.firebase.storage.ktx)
```
Comment (chenyair) [inline android/app/build.gradle.kts:73]: Commit your libs.version.toml file also

File: android/app/src/main/res/layout/fragment_add_experience.xml [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+                    app:layout_constraintTop_toTopOf="parent" />
+
+                <LinearLayout
+                    android:id="@+id/llNoSelectionPlaceholder"
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:alpha="0.16"
+                    android:gravity="center"
+                    android:orientation="vertical"
+                    android:visibility="invisible">
```
Comment (chenyair) [inline android/app/src/main/res/layout/fragment_add_experience.xml:95]: It is always invisible. In the code section I only see the state being change to gone but its never visible

File: android/app/src/main/res/layout/fragment_add_experience.xml [hunk 7]
```diff
@@ -0,0 +1,251 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (chenyair) [inline android/app/src/main/res/layout/fragment_add_experience.xml:2]: Why frame layout and not linear layout? Frame layout should only have one child and it looks like you got 2 here

**Comment by chenyair** [comment]
2 General bugs found:

- [ ] First time uploading an experience when opening the app, the redirection is to discover page instead of the back to the experience page
- [ ] After uploading one experience the upload button is enabled even when nothing is entered

File: android/app/src/main/res/layout/fragment_add_experience.xml (hunk 6 above)
Comment (liadyacobi) [reply android/app/src/main/res/layout/fragment_add_experience.xml:95]: I'll check, maybe this is a mistake and should be visible

**Comment by liadyacobi** [comment]
> 2 General bugs found:
> 
> * [ ]  First time uploading an experience when opening the app, the redirection is to discover page instead of the back to the experience page
> * [ ]  After uploading one experience the upload button is enabled even when nothing is entered

Referenced and fixed in #132

**Pull request merged**